Celeste.Mod.mm/Mod/UI/TextMenuExt.cs

```diff
@@ -355,5 +357,879 @@ public class IntSlider : TextMenu.Item {
+                }
+                do {
+                    Selection += direction;
+                    if (num > 2) {
+                        if (Selection < 0) {
+                            Selection = Items.Count - 1;
+                        } else if (Selection >= Items.Count) {
+                            Selection = 0;
+                        }
+                    } else if (Selection < 0 || Selection > Items.Count - 1) {
+                        Selection = Calc.Clamp(Selection, 0, Items.Count - 1);
+                        break;
+                    }
+                }
+                while (!Current.Hoverable);
+
+                if (!Current.Hoverable) {
+                    Selection = selection;
+                }
+                if (Selection != selection && Current != null) {
+                    if (selection >= 0 && Items[selection] != null && Items[selection].OnLeave != null) {
+                        Items[selection].OnLeave();
+                    }
+                    Current.OnEnter?.Invoke();
+                    if (wiggle) {
+                        Audio.Play(direction > 0 ? "event:/ui/main/rollover_down" : "event:/ui/main/rollover_up");
+                        Current.SelectWiggler.Start();
+                    }
+                }
+            }
+
+            public void RecalculateSize() {
+                TitleHeight = ActiveFont.LineHeight;
+                if (Items.Count < 1)
+                    return;
+
+                LeftColumnWidth = RightColumnWidth = MenuHeight = 0f;
+                foreach (TextMenu.Item item in Items) {
+                    if (item.IncludeWidthInMeasurement) {
+                        LeftColumnWidth = Math.Max(LeftColumnWidth, item.LeftWidth());
+                    }
+                }
+                foreach (TextMenu.Item item in Items) {
+                    if (item.IncludeWidthInMeasurement) {
+                        RightColumnWidth = Math.Max(RightColumnWidth, item.RightWidth());
+                    }
+                }
+                foreach (TextMenu.Item item in Items) {
+                    if (item.Visible) {
+                        MenuHeight += item.Height() + Container.ItemSpacing;
+                    }
+                }
+                MenuHeight -= Container.ItemSpacing;
+            }
+
+            public float GetYOffsetOf(TextMenu.Item item) {
+                if (item == null) {
+                    return 0f;
+                }
+                float offset = 0f;
+                foreach (TextMenu.Item item2 in Items) {
+                    if (item2.Visible) {
+                        offset += item2.Height() + ItemSpacing;
+                    }
+                    if (item2 == item) {
+                        break;
+                    }
+                }
+                return offset - item.Height() * 0.5f - ItemSpacing + Container.GetYOffsetOf(this);
+            }
+
+            private void menu_Added() {
```

Comment:
Can be merged into non-menu_ method.

Reply:
I was originally doing this in order to keep code related to the item and menu as separate as possible. Should I still just merge them together?